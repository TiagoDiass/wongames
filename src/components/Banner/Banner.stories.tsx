import { ComponentMeta, ComponentStory } from '@storybook/react';
import Banner from './Banner';

export default {
  title: 'Banner',
  component: Banner,
  args: {
    // img: 'https://source.unsplash.com/user/willianjusten/1042x580',
    img: 'https://images.unsplash.com/photo-1604818659418-1c53672b00f6?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=580&ixid=MnwxfDB8MXxyYW5kb218MHx8Y3liZXJwdW5rfHx8fHx8MTY0ODE2NTgwMQ&ixlib=rb-1.2.1&q=80&utm_campaign=api-credit&utm_medium=referral&utm_source=unsplash_source&w=1042',
    title: 'Defy Death',
    subtitle: '<p>Play the new <strong>CrashLands</strong> season',
    buttonLabel: 'Buy now',
    buttonLink: '/games/defy-death'
  },
  parameters: {
    layout: 'fullscreen'
  }
} as ComponentMeta<typeof Banner>;

export const Basic: ComponentStory<typeof Banner> = (args) => <Banner {...args} />;
