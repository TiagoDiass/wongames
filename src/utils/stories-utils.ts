type Backgrounds = 'Dark' | 'Light' | 'Light Blue';

export const setStoryDefaultBackground = (background: Backgrounds) => ({
  backgrounds: {
    default: background
  }
});

type Layouts = 'fullscreen' | 'centered' | 'padded';

export const setStoryLayout = (storyLayout: Layouts) => ({
  layout: storyLayout
});
