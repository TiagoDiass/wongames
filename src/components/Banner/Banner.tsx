import * as S from './Banner.styles';
import { Button } from 'components';

export type BannerProps = {
  img: string;
  title: string;
  subtitle: string;
  buttonLabel: string;
  buttonLink: string;
};

/**
 * Component that will contains a Banner/Carousel
 */
export default function Banner({ img, title, subtitle, buttonLabel, buttonLink }: BannerProps) {
  return (
    <S.Wrapper>
      <S.Image src={img} role='img' aria-label={title} />

      <S.Caption>
        <S.Title>{title}</S.Title>
        <S.Subtitle dangerouslySetInnerHTML={{ __html: subtitle }} />

        <Button as='a' href={buttonLink} size='large'>
          {buttonLabel}
        </Button>
      </S.Caption>
    </S.Wrapper>
  );
}
