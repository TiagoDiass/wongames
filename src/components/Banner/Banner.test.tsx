import Banner from './Banner';
import { renderWithTheme } from 'utils/test-utils';
import { screen } from '@testing-library/react';

describe('Component: Banner', () => {
  it('should render correctly', () => {
    const BANNER_PROPS = {
      img: 'https://source.unsplash.com/user/willianjusten/1042x580',
      title: 'Defy Death',
      subtitle: '<p>Play the new <strong>CrashLands</strong> season',
      buttonLabel: 'Buy now',
      buttonLink: '/games/defy-death'
    } as const;

    const { container } = renderWithTheme(<Banner {...BANNER_PROPS} />);

    expect(screen.getByRole('heading', { name: BANNER_PROPS.title })).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: 'Play the new CrashLands season' })
    ).toBeInTheDocument();
    expect(screen.getByRole('link', { name: BANNER_PROPS.buttonLabel })).toHaveAttribute(
      'href',
      BANNER_PROPS.buttonLink
    );
    expect(screen.getByRole('img', { name: BANNER_PROPS.title })).toHaveAttribute(
      'src',
      BANNER_PROPS.img
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
