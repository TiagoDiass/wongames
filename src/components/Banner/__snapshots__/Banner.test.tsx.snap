// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Banner should render correctly 1`] = `
.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-color: #EAEAEA;
  background-image: url(https://source.unsplash.com/user/willianjusten/1042x580);
  background-position: center center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 2.4rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 400;
  margin-bottom: 1.6rem;
}

.c4 strong {
  color: #F231A5;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  border-radius: 0.4rem;
  outline: none;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c5:disabled {
  opacity: 0.7;
  cursor: not-allowed;
}

.c5:disabled:hover {
  -webkit-filter: none;
  filter: none;
}

.c5:focus {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c5:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

@media (max-width:1170px) {

}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
  }
}

@media (min-width:768px) {
  .c1 {
    height: 58rem;
  }
}

@media (min-width:768px) {
  .c2 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

<div
  class="c0"
>
  <div
    aria-label="Defy Death"
    class="c1"
    role="img"
    src="https://source.unsplash.com/user/willianjusten/1042x580"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      Defy Death
    </h2>
    <h3
      class="c4"
    >
      <p>
        Play the new 
        <strong>
          CrashLands
        </strong>
         season
      </p>
    </h3>
    <a
      class="c5"
      href="/games/defy-death"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</div>
`;
